docs/zkapps/zkapp-development-frameworks.mdx

```diff
@@ -15,68 +15,214 @@ keywords:
-  - Where more than the eight on-chain field elements are required to manage state, and access to that state is not shared between users, the experimental [Offchain Storage API](/zkapps/writing-a-zkapp/feature-overview/offchain-storage) offers a solution.
-  - Where concurrent access to _shared global state_ is required, the required architecture is available **out of the box** when using the Protokit framework to build your zkApp as an zkApp-chain (L2). There is currently no easy-to-use equivalent for shared state in o1js L1 contracts.
 
-Start here:
+## o1js[​](https://docs.minaprotocol.com/zkapps/zkapp-development-frameworks#o1js)
 
-- [Developer documentation](/zkapps/o1js)
-- [o1js repository](https://github.com/o1-labs/o1js)
-- [Discord](https://discord.gg/minaprotocol)
+[o1js](https://docs.minaprotocol.com/zkapps/o1js) is the best way to author **highly-optimized ZK circuits and primitives**, but it does not provide an easy-to-use solution for managing a shared global state.
 
-## [Protokit](https://protokit.dev/)
+o1js can be used directly on the Mina L1, and is thus limited by the L1’s throughput and latency, but maintains strong decentralization guarantees.
 
-Protokit is a powerful framework designed to build **ZK appchains and smart contracts** that are user-facing, privacy-preserving, interoperable.  
-It offers a familiar developer experience similar to Solidity dApps, making it easier and intuitive for developers to leverage ZK in their blockchain solutions.
 
-It provides a full set of tools necessary for:
+<table>
+  <tr>
+   <td>Info
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Did you know?</strong> zkApps written directly with o1js can orchestrate concurrent users requiring shared state using Mina L1’s <a href="https://docs.minaprotocol.com/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer">Actions & Reducer</a>, but it requires significant development to orchestrate Action & Reducers. At its core, Protokit is a framework for orchestrating the Mina L1 Actions & Reducers out of the box.
+   </td>
+  </tr>
+</table>
+
+
+Start developing zkApps with o1js here:
+
+
+
+* [Developer documentation](https://docs.minaprotocol.com/zkapps/o1js)
+* [o1js repository](https://github.com/o1-labs/o1js)
+* [Discord](https://discord.gg/minaprotocol)
+
+
+## Choosing between Protokit and o1js
+
+**Still can’t decide between Protokit and o1js?** 
+
+This section provides a simple decision tree x`and comparison table to help zkApp developers choose which framework to use.
+
+![Decision Tree for zkApp devs](/img/decision-tree.png)
+
+Are you developing a zkApp with concurrent users requiring a shared state?
+1. Yes - Protokit
+    1. Are you willing to trade off L1 censorship resistance and liveness guarantees for high throughput and low latency?*
+        1. Yes - Hybrid mode
+        2. No - Based mode
+2. No - o1js
```

Comment:
Existence of `OffchainState` makes this tree a bit complicated than this. It's true that `OffchainState` is not scalable as base mode Protokit since it doesn't keep a db thus recreates the offchain state Merkle tree on-the-fly which creates an overhead. But, It's ready to use now (not production-ready) and worth mentioning in the tree.

Reply:
Considering this closed.